### PR TITLE
[#89017762] Another incremental step regarding authorization and proposals

### PR DESCRIPTION
**Pull request by cmc333333**
* Adds an authorization check for whether or not a user can see the approve/reject buttons (though nothing was added to the corresponding handler)
* That check allowed many decorator methods to be removed
* Streamlined 7 or so related but ever so slighty different ways of accessing approv(ers/als) into 4 entry points: all approv* and "current" approv*
* Defaulted approvals to ordering by the position field. This only needs to be overridden in one place, but removes a dozen "ordered" calls
* Moved `linear?` and `parallel?` to proposal

### CODE REVIEW

File: app/decorators/cart_decorator.rb [hunk 1]
```diff
@@ -10,7 +10,8 @@ def total_approvers
   end
 
   def approvals_by_status
-    object.approvals.order(
+    # Override default scope
+    object.approvals.reorder(
```
Comment (afeld) [inline app/decorators/cart_decorator.rb]: Had no idea that method existed!

File: app/policies/proposal_policy.rb [hunk 2]
```diff
@@ -21,6 +21,11 @@ def not_approved?
     !@proposal.approved?
   end
 
+  def approve_reject?
```
Comment (afeld) [inline app/policies/proposal_policy.rb]: What do you think about prefixing the Policy method names with `can_`?

**Comment by afeld** [comment]
:clap:

File: app/policies/proposal_policy.rb (hunk 2 above)
Comment (cmc333333) [inline app/policies/proposal_policy.rb]: Makes sense, though the pundit people might take offense ;)

**Pull request merged**